src/bluesearch/mining/eval.py

```diff
@@ -362,6 +363,29 @@ def idx2text(tokens, idxs):
     )
 
 
+@overload
+def ner_report(
+    iob_true: pd.Series,
+    iob_pred: pd.Series,
+    mode: str = "entity",
+    etypes_map: Optional[dict] = None,
+    return_dict: Literal[False] = ...,
+) -> str:
+    ...
+
+
+@overload
+def ner_report(
+    iob_true: pd.Series,
+    iob_pred: pd.Series,
+    mode: str = "entity",
+    etypes_map: Optional[dict] = None,
+    *,
+    return_dict: Literal[True],
+) -> OrderedDict:
+    ...
+
+
```

Comment:
Hm, interesting - what's the background for this and what does it do?

Reply:
In several `eval.py` from `data_and_models` folder, we call this function `ner_report` and we go through the output `dictionary` thanks to `.items()`. However, from typing, `mypy` does not know if the output is a `str` or a `dict` and thus is complaining that `str` does not have any `items()` method. 

One solution for this, it is to use `cast()` (as we did in the past for `pandas`). However, in this specific case, we know from `return_dict` value what is going to be the output type. This part of the code is specifying that if `return_dict` is `True`, the output is going to be a `dict`, otherwise, it is going to be a `str`. Doing that, you do not need to cast your function anymore.

Reply:
Ah I see, thanks for the explanation!

To me this looks like a good reason for not using variable return types :)

Reply:
I think it is inspired by `sklearn.confusion_matrix`. However, I am against changing this part of the code to avoid variable return types. I don't think we have a lot of methods in this case except this one.

Reply:
No, no, of course not, I'm not suggesting we should refactor our code, that would clearly be outside of the scope of this PR. It was just a general observation :)